drivers/flash/nrf_qspi_nor.c

```diff
@@ -540,15 +540,71 @@ static int qspi_nor_read(struct device *dev, off_t addr, void *dest,
 	return rc;
 }
 
+/* addr aligned, sptr not null, slen less than 4 */
+static inline nrfx_err_t write_sub_word(struct device *dev, off_t addr,
+					const void *sptr, size_t slen)
+{
+	u8_t __aligned(4) buf[4];
+	nrfx_err_t res;
+
+	/* read out the whole word so that unchanged data can be
+	 * written back
+	 */
+	res = nrfx_qspi_read(buf, sizeof(buf), addr);
+	qspi_wait_for_completion(dev, res);
+
+	if (res == NRFX_SUCCESS) {
+		memcpy(buf, sptr, slen);
+		res = nrfx_qspi_write(buf, sizeof(buf), addr);
+		qspi_wait_for_completion(dev, res);
+	}
+
+	return res;
+}
+
+BUILD_ASSERT((CONFIG_NORDIC_QSPI_NOR_STACK_WRITE_BUFFER_SIZE % 4) == 0,
+	     "NOR stack buffer must be multiple of 4 bytes");
+
+#define NVMC_WRITE_OK ((CONFIG_NORDIC_QSPI_NOR_STACK_WRITE_BUFFER_SIZE - 0) > 0)
```

Comment:
I'm a little bit confused by this line. Care to elaborate why you need the `- 0` ?

Reply:
In this case we probably don't.  One would use `(SYM - 0)` in a situation where `SYM` may be undefined, because to the preprocessor it would evaluate to `( - 0)` making the condition valid and false.

Here I don't think the module can see a case where `CONFIG_NORDIC_QSPI_NOR_STACK_WRITE_BUFFER_SIZE` is not defined (and if it did the `BUILD_ASSERT` above would fail.  Gone now.